From f4342f9e537c02ef6e2f9b3da44ed83fc1fdb518 Mon Sep 17 00:00:00 2001
From: Rosen Penev <rosenp@gmail.com>
Date: Mon, 27 Jul 2026 15:17:28 -0700
Subject: [PATCH 08812/15333] ARM: dts: BCM5301X: Add missing phy-mode and
 fixed-link to DSA switch ports

Ports 5, 7 and 8 in the BCM5301x SRAB switch node carry an 'ethernet'
property linking them to GMACs, but lack the required 'phy-mode' and
'fixed-link' properties. The DSA port schema (dsa-port.yaml) mandates
that any port with an 'ethernet' property must also have 'phy-mode'
and one of 'fixed-link', 'phy-handle', or 'managed'.

Add phy-mode = "internal" and a fixed-link subnode to ports 5, 7
and 8, matching the properties already present on their respective
GMAC nodes (gmac0, gmac1, gmac2). This resolves dtbs_check schema
validation warnings for all BCM5301x-based board DTBs.

Assisted-by: Opencode:Big-Pickle
Signed-off-by: Rosen Penev <rosenp@gmail.com>
Link: https://lore.kernel.org/r/20260727221728.40811-1-rosenp@gmail.com
Signed-off-by: Florian Fainelli <florian.fainelli@broadcom.com>
---
 arch/arm/boot/dts/broadcom/bcm-ns.dtsi | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/arch/arm/boot/dts/broadcom/bcm-ns.dtsi b/arch/arm/boot/dts/broadcom/bcm-ns.dtsi
index 392a25713669..b671d3f45756 100644
--- a/arch/arm/boot/dts/broadcom/bcm-ns.dtsi
+++ b/arch/arm/boot/dts/broadcom/bcm-ns.dtsi
@@ -442,16 +442,29 @@ port@4 {
 			port@5 {
 				reg = <5>;
 				ethernet = <&gmac0>;
+				phy-mode = "internal";
+
+				fixed-link {
+					speed = <1000>;
+					full-duplex;
+				};
 			};
 
 			port@7 {
 				reg = <7>;
 				ethernet = <&gmac1>;
+				phy-mode = "internal";
+
+				fixed-link {
+					speed = <1000>;
+					full-duplex;
+				};
 			};
 
 			port@8 {
 				reg = <8>;
 				ethernet = <&gmac2>;
+				phy-mode = "internal";
 
 				fixed-link {
 					speed = <1000>;
-- 
2.55.0

